agent/grpc/handler.go

```diff
@@ -75,7 +75,7 @@ func (l *chanListener) Addr() net.Addr {
 }
 
 func (l *chanListener) Close() error {
-	close(l.conns)
+	close(l.done)
```

Comment:
I don't think it's bad this way, but was there a specific problem with relying on closing the conns chan as a stop signal rather than a separate explicit done chan?

Reply:
Ya, I isolated this change in 2d70d24218d631fb273ed45e60410eae27bea5e6:

Closing l.conns can lead to a race and a 'panic: send on closed chan' when a connection is in the middle of being handled when the server is shutting down.

Found using `-race -count=800`.